src/transformers/models/efficientloftr/modeling_efficientloftr.py

```diff
@@ -0,0 +1,1446 @@
+                    current_channel_dim,
+                    out_channels,
+                    stride,
+                )
+            )
+            current_channel_dim = out_channels
+        self.blocks = nn.ModuleList(blocks)
+
+    def forward(
+        self, hidden_states: torch.Tensor, output_hidden_states: Optional[bool] = False
+    ) -> Tuple[torch.Tensor, Optional[Tuple[torch.Tensor]]]:
+        all_hidden_states = () if output_hidden_states else None
+        for block in self.blocks:
+            hidden_states = block(hidden_states)
+            if output_hidden_states:
+                all_hidden_states = all_hidden_states + (hidden_states,)
+        return hidden_states, all_hidden_states
+
+
+class EfficientLoFTRepVGG(nn.Module):
+    def __init__(self, config: EfficientLoFTRConfig):
+        super().__init__()
+
+        self.stages = nn.ModuleList([])
+        num_stages = len(config.stage_block_dims)
+        current_in_channels = 1
+
+        for i in range(num_stages):
+            out_channels = int(config.stage_block_dims[i] * config.stage_hidden_expansion[i])
+            stage = EfficientLoFTRRepVGGStage(
+                config, current_in_channels, out_channels, config.stage_num_blocks[i], config.stage_stride[i]
+            )
+            current_in_channels = out_channels
+            self.stages.append(stage)
+
+    def forward(
+        self, hidden_states: torch.Tensor, output_hidden_states: Optional[bool] = False
+    ) -> Tuple[List[torch.Tensor], Optional[Tuple[torch.Tensor]]]:
+        outputs = []
+        all_hidden_states = () if output_hidden_states else None
+        for stage in self.stages:
+            stage_outputs = stage(hidden_states, output_hidden_states=output_hidden_states)
+            hidden_states = stage_outputs[0]
+            if output_hidden_states:
+                all_hidden_states = all_hidden_states + stage_outputs[1]
+            outputs.append(hidden_states)
+
+        # Exclude first stage in outputs
+        outputs = outputs[1:]
+        return outputs, all_hidden_states
+
+
+class EfficientLoFTRAggregationLayer(nn.Module):
+    def __init__(self, config: EfficientLoFTRConfig):
+        super().__init__()
+
+        hidden_size = config.hidden_size
+        aggregation_sizes = config.aggregation_sizes
+        self.q_aggregation = (
+            nn.Conv2d(
+                hidden_size,
+                hidden_size,
+                kernel_size=aggregation_sizes[0],
+                padding=0,
+                stride=aggregation_sizes[0],
+                bias=False,
+                groups=hidden_size,
+            )
+            if aggregation_sizes[0] != 1
+            else nn.Identity()
+        )
+
+        if aggregation_sizes[1] != 1:
+            self.kv_aggregation = torch.nn.MaxPool2d(kernel_size=aggregation_sizes[1], stride=aggregation_sizes[1])
+        else:
+            self.kv_aggregation = nn.Identity()
+
+        self.norm = nn.LayerNorm(hidden_size)
```

Comment:
A bit sceptic about this code path, we have [4, 4] in config and Identity is never used.

```suggestion
        if aggregation_sizes[0] != 1:
            self.q_aggregation = nn.Conv2d(
                hidden_size,
                hidden_size,
                kernel_size=aggregation_sizes[0],
                padding=0,
                stride=aggregation_sizes[0],
                bias=False,
                groups=hidden_size,
            )
        else:
            self.q_aggregation = nn.Identity()

        if aggregation_sizes[1] != 1:
            self.kv_aggregation = torch.nn.MaxPool2d(kernel_size=aggregation_sizes[1], stride=aggregation_sizes[1])
        else:
            self.kv_aggregation = nn.Identity()

        self.norm = nn.LayerNorm(hidden_size)
```

Reply:
I get it, it is always hard to find the best tradeoff between using parameter as if other options were possible (which we can't tell apart from new EfficientLoFTR version that may be release in the future, if any) and getting rid of it. Do you think I should get rid of this parameter and just hardcode it ? 🤔

Reply:
Ok, let's keep it, but rename `aggregation_sizes` to `q_aggregation_kernel_size` and `kv_aggregation_kernel_size`

Reply:
And the same for strides

Reply:
Done !